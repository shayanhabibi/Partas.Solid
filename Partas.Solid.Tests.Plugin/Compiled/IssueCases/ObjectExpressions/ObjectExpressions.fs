module Partas.Solid.Tests.IssueCases.ObjectExpressions

open Partas.Solid
open Fable.Core
open Fable.Core.JS
open Fable.Core.JsInterop

[<AllowNullLiteral; Interface>]
type AbstractInterface =
    abstract member delay: int with get, set

[<Erase>]
type WordRotate() =
    inherit div()

    [<Erase>]
    member val words: string = unbox null with get, set

    [<Erase>]
    member val duration: int = unbox null with get, set

    [<SolidTypeComponentAttribute(ComponentFlag.DebugMode)>]
    member props.constructor =
        let index, setIndex = createSignal (0)

        let transition =
            jsOptions<AbstractInterface> (fun o ->
                o.delay <-
                    props.duration
                    * index ())

        For (each = props.words.Split (' '))
