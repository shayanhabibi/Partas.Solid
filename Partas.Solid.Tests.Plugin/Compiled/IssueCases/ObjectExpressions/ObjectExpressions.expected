import { For, createSignal, splitProps } from "solid-js";
import { split } from "../../fable_modules/fable-library-js.5.0.0-alpha.13/String.js";

export function WordRotate(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["duration", "words"]);
    const patternInput = createSignal(0);
    const transition = {
        delay: PARTAS_LOCAL.duration * patternInput[0](),
    };
    return <For each={split(PARTAS_LOCAL.words, [" "], undefined, 0)} />;
}

